src/smt/preprocessor.cpp

```diff
@@ -140,7 +143,8 @@ Node Preprocessor::simplify(const Node& node, bool removeItes)
   }
   std::unordered_map<Node, Node, NodeHashFunction> cache;
   Node n = d_processor.expandDefinitions(nas, cache);
-  Node ns = applySubstitutions(n);
+  TrustNode ts = d_ppContext->getTopLevelSubstitutions().apply(node);
+  Node ns = ts.isNull() ? node : ts.getNode();
```

Comment:
I guess we should use `n` instead of `node`?
Also: why is the rewrite gone (that was there in `applySubstitutions()`)?

Reply:
Yes, you're right about `n`, good catch.   Rewrite is not here since it is done within TrustSubstitutionMap.